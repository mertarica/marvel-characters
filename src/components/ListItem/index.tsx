import { ICharacter } from "../../types/character.type";
import { Link } from "react-router-dom";

type Props = {
  character: ICharacter;
};

const ListItem = ({ character }: Props) => {
  return (
    <div className="c-list-item" data-testid="list-item">
      <Link to={`/details/${character.id}`}>
        <div className="c-list-item__image">
          <img
            src={`${character.thumbnail?.path}.${character.thumbnail?.extension}`}
            alt={character.name}
            className="w-100"
          />
        </div>
        <div className="c-list-item__info">
          <h5 className="c-list-item__info-header">{character.name}</h5>
        </div>
      </Link>
    </div>
  );
};

export default ListItem;
