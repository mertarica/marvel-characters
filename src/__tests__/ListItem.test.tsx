import renderer from "react-test-renderer";
import ListItem from "../components/ListItem";
import fakeCharacter from "../__mock__/character";

describe("<ListItem />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ListItem character={fakeCharacter} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
