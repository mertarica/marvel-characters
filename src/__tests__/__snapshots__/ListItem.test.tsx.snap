// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> renders correctly 1`] = `
<div
  className="c-list-item"
  data-testid="list-item"
>
  <a
    href="/details/undefined"
    onClick={[Function]}
  >
    <div
      className="c-list-item__image"
    >
      <img
        alt="Fake Hero"
        className="w-100"
        src="."
      />
    </div>
    <div
      className="c-list-item__info"
    >
      <h5
        className="c-list-item__info-header"
      >
        Fake Hero
      </h5>
    </div>
  </a>
</div>
`;
